test/membrane_live_web/plug/auth_plug_integration_test.exs

```diff
@@ -0,0 +1,112 @@
+      %{endpoint_atom: :delete},
+      %{endpoint_atom: :show},
+      %{endpoint_atom: :index}
+    ]
+
+    Enum.map(@webinar_endpoint_infos, fn %{endpoint_atom: atom} = info ->
+      test "[401] #{atom} webinar", %{conn: conn_with_wrong_bearer} do
+        test_webinar_endpoint(conn_with_wrong_bearer, unquote(Macro.escape(info)))
+      end
+    end)
+
+    defp test_webinar_endpoint(conn, %{endpoint_atom: atom} = endpoint_info)
+         when atom in @arity_2_endpoints do
+      test_endpoint(&Routes.webinar_path/2, conn, endpoint_info)
+    end
+
+    defp test_webinar_endpoint(conn, %{endpoint_atom: atom} = endpoint_info)
+         when atom in @arity_3_endpoints do
+      test_endpoint(&Routes.webinar_path/3, conn, endpoint_info)
+    end
+  end
+
+  describe "user" do
+    @user_endpoint_infos [
+      %{endpoint_atom: :update, body: %{user: user_attrs()}},
+      %{endpoint_atom: :delete},
+      %{endpoint_atom: :show},
+      %{endpoint_atom: :index}
+    ]
+
+    Enum.map(@user_endpoint_infos, fn %{endpoint_atom: atom} = info ->
+      test "[401] user #{atom} resources test", %{conn: conn_with_wrong_bearer} do
+        test_user_endpoint(conn_with_wrong_bearer, unquote(Macro.escape(info)))
+      end
+    end)
+
+    defp test_user_endpoint(conn, %{endpoint_atom: atom} = endpoint_info)
+         when atom in @arity_2_endpoints do
+      test_endpoint(&Routes.user_path/2, conn, endpoint_info)
+    end
+
+    defp test_user_endpoint(conn, %{endpoint_atom: atom} = endpoint_info)
+         when atom in @arity_3_endpoints do
+      test_endpoint(&Routes.user_path/3, conn, endpoint_info)
+    end
+  end
+
+  defp test_endpoint(routing_function, conn, %{endpoint_atom: atom} = endpoint_info)
+       when atom in @endpoint_atoms do
+    req_args = compute_req_args(routing_function, conn, endpoint_info)
+
+    atom
+    |> endpoint_atom_to_function()
+    |> apply(req_args)
+    |> assert_status_is_401()
+  end
+
+  defp compute_req_args(routing_function, conn, %{endpoint_atom: atom, body: body}) do
+    [conn, compute_routing_function(routing_function, conn, atom), body]
+  end
+
+  defp compute_req_args(routing_function, conn, %{endpoint_atom: atom}) do
+    [conn, compute_routing_function(routing_function, conn, atom)]
+  end
+
+  defp compute_routing_function(routing_function, conn, atom) do
+    case :erlang.fun_info(routing_function)[:arity] do
+      2 -> routing_function.(conn, atom)
+      3 -> routing_function.(conn, atom, @dummy_uuid)
+    end
+  end
+
+  # since we return a function, this cannot be converted to a module attribute map
+  defp endpoint_atom_to_function(:create), do: &post/3
+  defp endpoint_atom_to_function(:update), do: &put/3
+  defp endpoint_atom_to_function(:index), do: &get/2
+  defp endpoint_atom_to_function(:show), do: &get/2
+  defp endpoint_atom_to_function(:delete), do: &delete/2
+
+  defp assert_status_is_401(conn), do: conn |> response(401) |> assert()
```

Comment:
This also could be moved to `test_endpoint`.

Reply:
I've left this that way. I can change it to `auth_assert` and add further checks